### DISCUSSION TITLE
How to install extensions in WebView2 prerelease v1.0.2065?

### DISCUSSION THREAD

**Discussion by Nocturna22** [MicrosoftEdge/WebView2Feedback#4061] [Q&A]
Hello, currently I am trying to install an extension. I try to check if extensions are allowed, but I do not find the function.
So I tried to install the extension:

`WebView2_Main.CoreWebView2.Profile.AddBrowserExtensionAsync("C:\Users\xxxx\Desktop\xxxx\chrome\xx-xxt-xxx\xxxxxx_v7.1.zip")`

I also tried unzipping the zip file and specified the folder. There is a manifest.json in there.
Also I tried to get a list of all installed extensions:

`MsgBox(WebView2_Main.CoreWebView2.Profile.GetBrowserExtensionsAsync.ToString)`

But that didn't work either...
So how do you install an extension?
And in which path is the extension installed then?

**Comment by mikeduglas** [comment]
I'm able to install extensions. First, you must set CoreWebView2EnvironmentOptions.AreBrowserExtensionsEnabled to true. Then, call CoreWebView2.Profile.AddBrowserExtensionAsync("folder containing manifest.json").

**Comment by Nocturna22** [comment]
Thank you very much. I use the control via the designer. I have some difficulties to set the options of the control.

did you initialize the control via code, or via the designer?

**Comment by mikeduglas** [comment]
via code, like this:
```
CoreWebView2EnvironmentOptions options = new CoreWebView2EnvironmentOptions(customSchemeRegistrations: csRegistrations);
options.AreBrowserExtensionsEnabled = true;

CoreWebView2Environment env = await CoreWebView2Environment.CreateAsync(
  cp?.BrowserExecutableFolder,
  cp?.UserDataFolder,
  options);

await webView.EnsureCoreWebView2Async(env);
```

**Comment by Nocturna22** [comment]
Yes this looks like my code.
```

Dim options As CoreWebView2EnvironmentOptions = New CoreWebView2EnvironmentOptions()
options.AreBrowserExtensionsEnabled = True
Dim env As CoreWebView2Environment = Await CoreWebView2Environment.CreateAsync(Nothing, Nothing, options)


        Me.Show()

        Do While Not waitToLoad
            Application.DoEvents()
            ' Do nothing
        Loop

        WebView2_Main.Visible = True
        Await WebView2_Main.EnsureCoreWebView2Async(env)
```

The problem is that I just dragged and dropped the control onto the form and did not initialize it via code. It must be possible to set the options this way. If I do it as described above, I get an error message that the element has already been initialized. I have looked in the designer. I do not know where I have to insert that there, if it goes at all ^^ I mean, there it must be initialized, however, you can fit there nowhere the env



`  Private Sub InitializeComponent()
      Dim resources As ComponentModel.ComponentResourceManager = New ComponentModel.ComponentResourceManager(GetType(Form_Main))

      WebView2_Main = New Microsoft.Web.WebView2.WinForms.WebView2()
      Button1 = New Button()
      CType(WebView2_Main, ComponentModel.ISupportInitialize).BeginInit()
      SuspendLayout()
      ' 
      ' WebView2_Main
      ' 


      WebView2_Main.AllowExternalDrop = True
      WebView2_Main.BackColor = Color.Black
      WebView2_Main.BackgroundImageLayout = ImageLayout.Center
      WebView2_Main.CreationProperties = Nothing
      WebView2_Main.DefaultBackgroundColor = Color.Black
      WebView2_Main.Dock = DockStyle.Fill
      WebView2_Main.Location = New Point(0, 0)
      WebView2_Main.Name = "WebView2_Main"
      WebView2_Main.Size = New Size(1484, 861)
      WebView2_Main.Source = New Uri("https://someurl.com", UriKind.Absolute)
      WebView2_Main.TabIndex = 0
      WebView2_Main.Visible = False
      WebView2_Main.ZoomFactor = 1.0R`


Almost looks like I have to create the WebView2 element manually -.-"

**Comment by Monarkos** [comment]
How to do this in MAUI C#?
